### PR TITLE
shadow-bot 1.4.1 (new cask)

### PR DESCRIPTION
**Pull request by moonfruit**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/s/shadow-bot.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+cask "shadow-bot" do
+  arch arm: "-arm64", intel: ""
```
Comment (bevanjkay) [inline Casks/s/shadow-bot.rb:2]:
```suggestion
  arch arm: "-arm64"
  livecheck_query = on_arch_conditional arm: "arm64", intel: "intel"
```

Accompanied by changes to `livecheck` below.

File: Casks/s/shadow-bot.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  url "https://winrobot-pub-a-cos.winrobot360.com/client/install/ShadowBot-#{version}#{arch}.dmg",
+      verified: "winrobot-pub-a-cos.winrobot360.com/"
+  name "Shadow Bot"
+  name "影刀"
+  desc "Application for robotic process automation"
+  homepage "https://www.yingdao.com/"
+
+  livecheck do
+    url "https://api.yingdao.com/api/v1/sys/download/client?os=mac&arch=intel"
```
Comment (bevanjkay) [inline Casks/s/shadow-bot.rb:16]:
```suggestion
  url "https://api.yingdao.com/api/v1/sys/download/client?os=mac&arch=#{livecheck_query}"
  regex(/ShadowBot[._-]v?(\d+(?:\.\d+)+)#{arch}\.dmg/i)
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]
Please see review comments above.
Does the app have an in-app updating mechanism?

**Comment by moonfruit** [comment]
I'm not aware of any in-app updating mechanism for now.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**